mesonbuild/build.py

```diff
@@ -538,13 +539,9 @@ def __post_init__(self, overrides: T.Optional[T.Dict[OptionKey, str]]) -> None:
         else:
             ovr = {}
         self.options = OptionOverrideProxy(ovr, self.environment.coredata.options, self.subproject)
-        # XXX: this should happen in the interpreter
         if has_path_sep(self.name):
-            # Fix failing test 53 when this becomes an error.
-            mlog.warning(textwrap.dedent(f'''\
-                Target "{self.name}" has a path separator in its name.
-                This is not supported, it can cause unexpected failures and will become
-                a hard error in the future.'''))
+            self.install_name = os.path.basename(self.name)
+            self.name = flatten_path(self.name)
```

Comment:
I'm nervous about automatically making this change...

Reply:
instead of 'flatten_path', perhaps some other mechanism which generated unique prefixes could be used to ensure that the resulting 'name' didn't conflict with anything else in the project? It would be odd for a project to include both foo/bar and foo_bar, but it could happen.

Reply:
oh, and that's why I split this bit out into a separate patch -- I'd be fine with just adding support for the new property and then just waiting for that code to become available before using it.

Reply:
I've put the warning message back in so that if it breaks, users will at least know it's their fault?